packages/components/src/useSteps.js

```diff
@@ -1,13 +1,16 @@
-import { useContext, useMemo } from 'react'
-import { Context } from './context'
+import { useContext, useEffect } from 'react'
+import { Context as C } from './context'
 
-export default length => {
+export const useStepsFactory = Context => length => {
```

Comment:
I think the mdx-deck context should be the only context used here – feel free to copy/fork the code if you want this functionality elsewhere

Reply:
This was meant to make the hook testable by its self and without having to mock out the context. This way you can pass in a context that's contained in the test.